coderd/apikey.go

```diff
@@ -516,3 +604,65 @@ func (api *API) createAPIKey(ctx context.Context, params apikey.CreateParams) (*
 		HttpOnly: true,
 	}), &newkey, nil
 }
+
+func normalizeTokenScopes(scope codersdk.APIKeyScope, scopes []codersdk.APIKeyScope) (database.APIKeyScopes, error) {
+	// Default to coder:all for backward compatibility when nothing is provided.
+	if len(scopes) == 0 && string(scope) == "" {
+		return database.APIKeyScopes{database.ApiKeyScopeCoderAll}, nil
+	}
+
+	if len(scopes) == 0 {
+		name := string(scope)
+		if name == "" {
+			return database.APIKeyScopes{database.ApiKeyScopeCoderAll}, nil
+		}
+		if !rbac.IsExternalScope(rbac.ScopeName(name)) {
+			return nil, xerrors.Errorf("invalid or unsupported API key scope: %q", name)
+		}
+		switch name {
+		case "all":
+			return database.APIKeyScopes{database.ApiKeyScopeCoderAll}, nil
+		case "application_connect":
+			return database.APIKeyScopes{database.ApiKeyScopeCoderApplicationConnect}, nil
+		default:
+			return database.APIKeyScopes{database.APIKeyScope(name)}, nil
+		}
+	}
+
+	out := make(database.APIKeyScopes, 0, len(scopes))
+	for _, raw := range scopes {
+		name := string(raw)
+		if !rbac.IsExternalScope(rbac.ScopeName(name)) {
+			return nil, xerrors.Errorf("invalid or unsupported API key scope: %q", name)
+		}
+		out = append(out, database.APIKeyScope(name))
+	}
+	return out, nil
+}
+
+func normalizeTokenAllowList(entries []codersdk.APIAllowListTarget) (database.AllowList, bool, error) {
```

Comment:
Do we need the `bool` return? It just returns if the input list is non-zero right?

It's only used here
https://github.com/coder/coder/blob/2a701ed574f1c844d537039ec31abf55131dd018/coderd/apikey.go#L92-L101

But the `params.AllowList` is not set if `provided == false`. So the boolean return does not do anything.